### DISCUSSION TITLE
Feature Request: [Spatial] Utilize HTTP Range-Request in remote FlatGeobuf fetching

### DISCUSSION THREAD

**Discussion by Kanahiro** [duckdb/duckdb#10088] [Ideas]
FlatGeobuf: https://flatgeobuf.org/

FlatGeobuf(FGB) is one of the cloud-native formats.
It is single file format but it has a spatial index and we can fetch partial contents of it via HTTP Range-Request.
example of range-request: https://flatgeobuf.org/examples/maplibre/

In duckdb, we can use FGB with spatial extension:

```sh
INSTALL spatial;
LOAD spatial;
select * from '/vsicurl/https://flatgeobuf.org/test/data/UScounties.fgb';
```

The size of this FGB is about 13MB so we can retrieve entire file in short time but there are larger data.
example: https://flatgeobuf.septima.dk/population_areas.fgb (12GB)

Currently we always have to retrieve entire file but FGB supports HTTP Range-Request so it is great to fetch partial data of FGB when spatial filter is active like:

```sh
select * from '/vsicurl/https://-------/somedata.fgb' WHERE ST_Intersects(geom, area_of_interest);
```

How do you think about this?

**Comment by Maxxen** [comment]
Hi! 

This is already possible if you use the st_read function and pass a spatial_filter_box argument. 

But in general we may be able to add an optimizer rule to look for a WHERE clause and "push it down" into the actual call to st_read (even when you just select from the file directly, internally its the same thing)

**Comment by Kanahiro** [comment]
Thanks for your kind response!
I could fetch patial contents with your advice, it's great! Example:

```sh
select * from ST_Read('/vsicurl/https://flatgeobuf.septima.dk/population_areas.fgb', spatial_filter_box=ST_Extent(ST_Buffer(ST_POINT(0, 0), 65.4)));
```

A process to make `BOX2D` is little a bit not easy to use but it is not big problem.
